src/library-authoring/collections/LibraryCollectionPage.tsx

```diff
@@ -0,0 +1,192 @@
+import { useContext, useEffect } from 'react';
+import { StudioFooter } from '@edx/frontend-component-footer';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import {
+  Badge,
+  Button,
+  Breadcrumb,
+  Container,
+  Icon,
+  IconButton,
+  Stack,
+} from '@openedx/paragon';
+import { Add, InfoOutline } from '@openedx/paragon/icons';
+import { Link, useParams } from 'react-router-dom';
+
+import Loading from '../../generic/Loading';
+import SubHeader from '../../generic/sub-header/SubHeader';
+import Header from '../../header';
+import NotFoundAlert from '../../generic/NotFoundAlert';
+import {
+  ClearFiltersButton,
+  FilterByBlockType,
+  FilterByTags,
+  SearchContextProvider,
+  SearchKeywordsField,
+  SearchSortWidget,
+} from '../../search-manager';
+import { useCollection, useContentLibrary } from '../data/apiHooks';
+import { LibraryContext } from '../common/context';
+import messages from './messages';
+import { LibrarySidebar } from '../library-sidebar';
+import LibraryCollectionComponents from './LibraryCollectionComponents';
+
+const HeaderActions = ({ canEditLibrary }: { canEditLibrary: boolean; }) => {
+  const intl = useIntl();
+  const {
+    openAddContentSidebar,
+  } = useContext(LibraryContext);
+
+  if (!canEditLibrary) {
+    return null;
+  }
+
+  return (
+    <div className="header-actions">
+      <Button
+        className="ml-1"
+        iconBefore={Add}
+        variant="primary rounded-0"
+        onClick={openAddContentSidebar}
```

Comment:
FYI there's a couple of changes required to this "add content" sidebar for when shown with a Collection:

* We can't add Collections to Collections, so the "add content sidebar" should hide or disable the "new Collection" button when shown here.
* New content objects need to be added to the Collection as well as its Library.

Reply:
@pomegranited Yes, I did not really do anything related to sidebars as part of this issue as I thought it is out of scope. 

Let me know if you want me to handle it either as part of this same task or a separate one.

Reply:
@navinkarkera Ya, this isn't mentioned in the scope of any existing tickets.. Do you have time to handle it here? If not, let me know, and I'll create a follow-up task.

Reply:
@pomegranited I can handle it here :+1:

Reply:
@pomegranited Updated to handle both cases but found one more case i.e. when pasting blocks in collection, it should be associated with the collection. I would prefer to handle this in a separate ticket if possible.